compiler/hash-ast/src/literal.rs

```diff
@@ -28,23 +26,25 @@ lazy_static! {
 impl StringLiteralMap {
     /// Add a new string to the map, this will add an additional entry even if the string is already
     /// within the map.
-    pub fn create_string(&self, value: AstString) -> StringIdentifier {
-        if let Some(key) = self.reverse_table.get(&value) {
+    pub fn create_string(&self, value: &str) -> StringLiteral {
+        if let Some(key) = self.reverse_table.get(value) {
             *key
         } else {
-            let ident = StringIdentifier::new();
+            let ident = StringLiteral::new();
 
             // copy over the string so that we can insert it into the reverse lookup table
-            let value_copy = value.clone();
+            // let wall = STATIC_CASTLE.wall();
+            // let value_copy = BrickString::new(value, &wall);
+            let value_copy = Box::leak(value.to_owned().into_boxed_str());
```

Comment:
Is this still copying over the string?

Reply:
Yes, which isn't necessary. I am also not sure as to whether to keep this as Box::leak or use a static castle. In essence, both leak memory, but the latter requires a murex lock every time.

Reply:
Keeping it as a `Box::leak` might be better if it avoids the mutex. Maybe it's better to put a `@@Cleanup` on this and leave it for another fight another day?